src/EquilibriumMeasures.jl

```diff
@@ -1,122 +1,187 @@
+
+import ForwardDiff: derivative, gradient, jacobian
+
+import LinearAlgebra: dot
+
+export equilibriummeasure, _equilibriummeasure, deflation_inner_products, deflation_deriv, deflation_op, deflation_scale
+
+Base.floatmin(::Type{<:ForwardDiff.Dual}) = floatmin(Float64)
+
+function _equilibriummeasure(V, a, b)
+    Typ = float(promote_type(typeof(a), typeof(b)))
+    T = ChebyshevT{Typ}()
+    x = Inclusion(a..b)
+    y = affine(x, axes(T,1)) # affine map from a..b to -1..1
+    T̃ = T[y,:]
+    wŨ = Weighted(ChebyshevU{Typ}())[y,:]
+    Ũ = ChebyshevU{Typ}()[y,:]
+    # Operators
+    H = T̃ \ (inv.(x .- x') * wŨ) # Hilbert wU\-> T̃
+    D = Ũ \ (Derivative(axes(T̃,1)) * T̃) # Derivative T̃ -> Ũ
+    ŨT̃ = Ũ \ T̃  # Converion T̃ -> Ũ
+    V_cfs = T̃ \ V.(x)
+    Vp_cfs = ŨT̃ \ (D * V_cfs)
+    Vp_cfs[1], wŨ * (H[2:end,:] \ Vp_cfs[2:end])/2
+end
+
+
+# Intentionally hide type for compile time
+struct EquilibriumMeasureMoment
+    V
+end
+
+function (E::EquilibriumMeasureMoment)(a) 
+    c0,μ = _equilibriummeasure(E.V, a...)
+    SVector(c0, sum(μ) - 1)
+end
+
+function equilibriummeasure(V; a = SVector(-1.0,1.0), maxiterations=1000, knownsolutions=[], power=2, shift=1.0, dampening=1.0, returnEndpoint=false)
+    μ = EquilibriumMeasureMoment(V)
+    num_found_sols = length(knownsolutions)
+    for k=1:maxiterations   
+        update = - jacobian(μ, a)\μ(a)
+        if num_found_sols > 0
+            update = deflation_scale(update, a, knownsolutions, power, shift) * update
+        end
+        an = a + dampening*update
+        if an ≈ a
+            # improve accuracy a bit more
+            for _=1:3
+                update = - jacobian(μ, a)\μ(a)
+                if num_found_sols > 0
+                    update = deflation_scale(update, a, knownsolutions, power, shift) * update
+                end
+                a = a + dampening*update
+            end
+            if returnEndpoint
+                return  _equilibriummeasure(V, a...)[2], a
+            else
+                return  _equilibriummeasure(V, a...)[2]
+            end
+        end
+        a = an
+    end
+    error("Max its")
+end
+
+# Deflation code
+
+function  deflation_inner_products(state::SVector, sol)
+    num_sols = length(sol)
+    T = promote_type(typeof(state[1]), typeof(sol[1][1]))
+
+    inner_products = zeros(T, num_sols)
+    for iter = 1:num_sols
+        inner_products[iter] = dot(state.-sol[iter], state.-sol[iter])
+    end
+    inner_products
+end
+
+function deflation_op(state::SVector, sol, power::Real, shift::Real)
+    num_sols = length(sol)
+    T = promote_type(typeof(state[1]), typeof(sol[1][1]))
+    power = T(power); shift = T(shift)
+
+    m = one(T)
+    inner_products = deflation_inner_products(state, sol)
+    
+    for iter = 1:num_sols
+        normsq = inner_products[iter]
+        factor = one(T)/normsq^(0.5*power) + shift

```

Comment:
```suggestion
        factor = one(T)/normsq^(power/2) + shift
```
`0.5` is a `Float64` so this version is more robust to other types.